torch/csrc/jit/autodiff.h

```diff
@@ -7,10 +7,59 @@
 
 namespace torch { namespace jit {
 
-// This function mutates the given graph (which should only contain a single stage)
-// by appending nodes of a next stage, which computes the Jacobian-vector product
-// (aka backward) of inputs to the first stage w.r.t. the outputs of the first stage.
-void differentiate(std::shared_ptr<Graph>& graph);
+struct Capture {
+  enum class Kind {Input, Output};
+  Capture(Kind kind, std::size_t offset)
+    : kind(kind), offset(offset) {}
+  Kind kind;
+  std::size_t offset;
+};
+
+using value_list = std::vector<Value*>;
+struct Gradient {
+  std::shared_ptr<Graph> f;
+  std::shared_ptr<Graph> df;
+
+  // Describes how to construct outputs of f from what its graph will return.
```

Comment:
These comments are good -- can you also copy the example from the pseudo-code into here as well?

Reply:
Hmm that will create a lot of duplication with the comment you requested. I'd rather put it in a single place, so I'll put it here as a note, and mention it over there.